File: 백준/Gold/1016. 제곱 ㄴㄴ 수/제곱 ㄴㄴ 수.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String [] args) throws IOException{
        BufferedReader br =new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        long min = Long.parseLong(st.nextToken());
        long max = Long.parseLong(st.nextToken());

        boolean[] check = new boolean[(int)(max-min)+1];

        int count=0;
        for(long i=2; i*i<=max; i++){
            long square = i*i;
            long start = min/square;
            if(min % square!=0){
                start+=1;
            }
            for(long j=start; j*square<=max; j++){
                check[(int)((j*square)-min)] = true;
            }
        }
        for(int i=0; i<=max-min; i++){
            if(!check[i]){
                count++;
            }
        }
        System.out.print(count);
    }
}

File: 백준/Gold/1016. 제곱 ㄴㄴ 수/제곱 ㄴㄴ 수.py
# 제곱이 아닌 수 찾기(골드1)
import math
import sys
input = sys.stdin.readline

a, b = map(int, input().split())
check=[True] *(b-a+1)

count=0
for i in range(2, int(math.sqrt(b))+1):
    pow = i*i
    start = math.ceil(a/pow)
    for j in range(start, int(b/pow)+1):
        check[j*pow-a] = False

for i in check:
    if i:
        count+=1
print(count)